autoscheduler/frontend/src/redux/actions/courseCards.ts

```diff
@@ -274,17 +286,32 @@ export function replaceCourseCards(
 ): ThunkAction<void, RootState, undefined, CourseCardAction> {
   // data for sections might have changed since last visit, so create a new course card
   // for each one asynchronously and update them with data from courseCards
-  return (dispatch): void => {
-    // clear all course cards before adding new ones
-    dispatch(clearCourseCards());
-
-    // if courseCards is improperly formatted or no cards are saved, do nothing
-    if (!Array.isArray(courseCards) || courseCards.length === 0) return;
+  return (dispatch, getState): void => {
+    // if courseCards is improperly formatted or no cards are saved, simply finish loading
+    if (!Array.isArray(courseCards) || courseCards.length === 0) {
+      dispatch(updateCourseCardSync(0, { loading: false }));
+      return;
+    }
 
-    // resolve promises for each course card and add them
+    // create course card from each serialized one, initially not fetching sections
+    const deserializedCards: CourseCardOptions[] = [];
     courseCards.forEach((courseCard, idx) => {
-      updateSectionsForCourseCard(courseCard, term).then((updatedCard) => {
-        dispatch(addCourseCard(updatedCard, idx));
+      const deserializedCard = deserializeCourseCard(courseCard);
+      deserializedCards.push(deserializedCard);
+      dispatch(updateCourseCardSync(idx, deserializedCard));
+    });
+
+    // all course cards are now loading (preventing courses from being overwritten
```

Comment:
```suggestion
    // all course cards are now marked as loading (preventing courses from being overwritten
```

slight clarification that would help me understand it if I ever came back to this code